# Problem 1 

In [53]:
#a list of numbers
aList = [1,5,7,9,20]

n = len(aList)
#print(n)

sm = 0
for x in aList:
    sm = sm + x
    num_sum = sm
#print(num_sum)

num_avg = num_sum/n
#print(num_avg)

var_sum = 0
for x in aList:
    y = (x - num_avg)* (x - num_avg)
    #print (y)
    var_sum = var_sum + y
#print(var_sum)

variance = (1/(n-1))*var_sum

print("The variance of a sample (aList) of",
      "n numbers is", variance)

The variance of a sample (aList) of n numbers is 50.800000000000004


In [42]:
import statistics

check = statistics.variance(aList)
print(check)

50.800000000000004


# Problem 2

In [64]:
def complement(seq):
    complement = {'A': 'T', 'T':'A', 'G':'C', 'C':'G'}
    bases = list(seq)
    bases = [complement[base] for base in bases]
    return ''.join(bases)
def reverse_complement(s):
    return complement(s[::-1])


print (reverse_complement("TACAGAT"))

ATCTGTA


# Problem 3

In [23]:
import random

sum_roll = 0
b = 0
avg_roll = 0

a = 1000
#a is number of simulated rolls

for x in range(a):
    #print(random.randint(1,6))
    x = random.randint(1,6)
    #print (x)
    sum_roll = sum_roll + x
    b = sum_roll

print("Total sum from 1000 rolls is", b)
avg_roll = b/a
print("Average from rolls is", avg_roll)

Total sum from 1000 rolls is 3474
Average from rolls is 3.474


This result does make sense, because each side of the die should have an equal chance of landing and the average of rolling all of those numbers once (1,2,3,4,5,6) is also 3.5.